Release `v0.17.0` (#2268)
valory-xyz/open-autonomy: docs/upgrading.md

# Open Autonomy

## `v0.16.1` to `v0.17.0`

> - The structure of the build folder when using Docker has been updated. 

Reviewer: I think it's updated for all deployment types
```suggestion
- The structure of the build folder has been updated. 
```

>   The folder is now named `abci_build_{index}`, 
>   where `{index}` is an incrementing number corresponding to the total number of existing builds, 

Author: ```suggestion
  The folder is now named `abci_build_{uuid}`, 
  where `{uuid}` is a 4-character random hash, 
```

>   where `{{index}}` is an incrementing number corresponding to the total number of existing builds, 

Reviewer: They are now like `abci_build_nki5`
```suggestion
  The folder is now named `abci_build_{{uuid}}`, 
  where `{{uuid}}` is a 4 character random hash., 
```

Author: Replaced with https://github.com/valory-xyz/open-autonomy/pull/2268#discussion_r1803012890 as this was marked as outdated after force pushing.

  replacing the previous format of `abci_build`.
- The naming convention for service containers has been revised. 
  Previously, containers were named `{service_name}_{abci|tm}_{index}`. 

>   Now, the format has changed to `{service_name}_{abci|tm}_{uuid}`, 

Reviewer: hey I just confirmed that these has actually become like `helloworldNfG2_tm_0`. So it should be like this right?
```suggestion
  Now, the format has changed to `{service_name}{uuid}_{abci|tm}_{index}`, 
```
I'm not sure because here the `service_name` is actually cleaned by removing _ and not exactly same as service name.

>   Previously, containers were named `{service_name}_{abci|tm}_{index}`. 
>   Now, the format has changed to `{service_name}{uuid}_{abci|tm}_{index}`, 
>   where `{uuid}` is a unique identifier.

Author: ```suggestion
  Previously, containers were named `{trimmed_service_name}_{abci|tm}_{index}`. 
  Now, the format has changed to `{trimmed_service_name}{uuid}_{abci|tm}_{index}`, 
  where `{uuid}` is a unique identifier.
```
